php/commands/src/CLI_Command.php

```diff
@@ -488,6 +488,20 @@ public function cmd_dump() {
 		echo json_encode( $this->command_to_array( EE::get_root_command() ) );
 	}
 
+	/**
+	 * Uninstalls easyengine completely along with all sites
+	 *
+	 * @subcommand self-uninstall
+	 */
+	public function self_uninstall() {
+		EE::confirm("Are you sure you want to remove EasyEngine and all its sites(along with their data) ?");
+		Utils\default_launch("docker rm -f $(docker ps -aqf label=org.label-schema.vendor=\"EasyEngine\")");
+		$home = Utils\get_home_dir();
+		Utils\default_launch("rm -rf $home/.ee/");
+		Utils\default_launch("rm -rf $home/ee-sites/");
+		Utils\default_launch("rm -rf " . EE_CONF_ROOT);
```

Comment:
What will happen if `EE_CONF_ROOT` is set to `/`? 🤔

Reply:
Since it's a constant used at many places I'd doubt we might let it happen 😅. But yes it seems a good idea to hardcode it.

Reply:
Done 👍

Reply:
What if a malicious actor executes ```EE_CONF_ROOT=/ ee cli self-uninstall``` 😉

Reply:
I think hardcoding it to `/opt/easyengine` is bad idea unless that path is going to be same always. Better approach is to add validation + avoid `rm -rf` (there must be a safer alternative)

Reply:
> I think hardcoding it to /opt/easyengine is bad idea unless that path is going to be same always.

It is always going to be same. We don't allow it to be changed.

> avoid rm -rf
another option could be not using `-rf` and instead manually remove each folder

```
rm /opt/easyengine/acme-conf/dir1/*
rm /opt/easyengine/acme-conf/dir2/*
rmdir /opt/easyengine/acme-conf/dir1/
rmdir /opt/easyengine/acme-conf/dir2/
rmdir /opt/easyengine/acme-conf/
```
and so on.

Reply:
> avoid rm -rf

You can use `\EE\Utils\delete_dir()` function or symfony filesystem `remove()`, that would be better than `rm -rf`.

Also, as @rahul286 suggested, the idea of adding validation is great, you can add the files and folders (which are limited and small in quantity, just the files and top level directories) that are present inside `/opt/easyengine` in a whitelisted array maybe and then check if they exist. Then move on to removing them. That array can have the files inside of `EE_CONF_ROOT` as well as the directories like `nginx` and `acme-conf`. Thus, if you check for these files inside and move on to removal only after you're sure that these files are present, you won't have to hardcode `/opt/easyengine`.

Reply:
I think removing whole `/opt/easyengine` makes sense since it'll always remain same and everything inside it will be related to easyengine anyways.